etc/ci_scripts/dump_artifacts/cli.py

```diff
@@ -21,33 +23,34 @@
 LOGGER.addHandler(logging.StreamHandler(stream=sys.stdout))
 
 
-def parse_args() -> Any:
-    import argparse
-
-    parser = argparse.ArgumentParser(
-        description="Dump all Docker logs for a given docker-compose project"
-    )
-    parser.add_argument(
-        "--compose-project",
-        dest="compose_project",
-        required=False,
-        default=None,
-        help="Docker Compose project. Do not specify if Docker-Compose is not involved (e.g. running against prod)",
-    )
-    parser.add_argument(
-        "--dump-agent-logs",
-        dest="dump_agent_logs",
-        required=False,
-        default=False,
-        help="Dump the logs for Nomad/Consul agents (e.g. running locally)",
-    )
-    return parser.parse_args()
+class Args:
+    def __init__(self) -> None:
+        parser = argparse.ArgumentParser(
+            description="Dump all Docker logs for a given docker-compose project"
+        )
+        parser.add_argument(
+            "--compose-project",
+            dest="compose_project",
+            required=False,
+            default=None,
+            help="Docker Compose project. Do not specify if Docker-Compose is not involved (e.g. running against prod)",
+        )
+        parser.add_argument(
+            "--dump-agent-logs",
+            dest="dump_agent_logs",
+            required=False,
+            default=False,
+            help="Dump the logs for Nomad/Consul agents (e.g. running locally). [True / False]",
+        )
+        args = parser.parse_args()
+        self.compose_project: Optional[str] = args.compose_project
+        # coerce from str -> Optional[bool] -> bool
+        self.dump_agent_logs = bool(to_bool(args.dump_agent_logs))
```

Comment:
personally, I really don't like argparse's bool handling, which is `--bool-flag` vs `--no-bool-flag`, preferring --dump-agent-logs=[Tt]rue. But the `--no-bool-flag` _is_ a posixism i think

Reply:
Counter argument: `--flag` vs `--no-flag` is inherently boolean, and you never have to wonder if the value is governed by `0` vs. `1`, `true` vs. `false`, `True` vs. `False`, `yes` vs. `no`, `Yes` vs. `No`, presence or absence, or any (potentially inconsistent!) subset of those options.

Reply:
fair. again, explicitly called this out knowing others would disagree, and my position isn't particularly defensible beyond personal preference, so I'll change it :)